src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -235,16 +235,18 @@ static void handle_resolution_exception(Symbol* class_name, bool throw_error, TR
   if (HAS_PENDING_EXCEPTION) {
     // If we have a pending exception we forward it to the caller, unless throw_error is true,
     // in which case we have to check whether the pending exception is a ClassNotFoundException,
-    // and if so convert it to a NoClassDefFoundError
-    // And chain the original ClassNotFoundException
+    // and convert it to a NoClassDefFoundError and chain the original ClassNotFoundException.
     if (throw_error && PENDING_EXCEPTION->is_a(SystemDictionary::ClassNotFoundException_klass())) {
       ResourceMark rm(THREAD);
       Handle e(THREAD, PENDING_EXCEPTION);
       CLEAR_PENDING_EXCEPTION;
       THROW_MSG_CAUSE(vmSymbols::java_lang_NoClassDefFoundError(), class_name->as_C_string(), e);
+    } else {
+      return; // the caller will throw the incoming exception
```

Comment:
Thanks for fixing this. It seems we must have some missing test coverage here though.

Reply:
The tests wouldn't have caught this since the VM ignored the new exception that this function tried to install below.